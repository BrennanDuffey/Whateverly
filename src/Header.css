header {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0px 190px;
  margin: 10px;
}

form {
  display: flex;
  padding-top: 28px;
}

input {
  -webkit-appearance: none;
  font-size: 1rem;
  height: 28px;
  width: 54%;
}

button {
  height: 27px;
}