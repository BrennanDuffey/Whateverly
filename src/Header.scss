header {
  background: rgb(53, 53, 53);
  box-shadow: 5px 15px 7px rgba(0, 0, 0, .5);
  color: #FCFDFF;
  display: flex;
  flex-direction: column;
  align-items: center;
  position: fixed;
  top: 0;
  width: 100%;
  padding: 10px 0;
}

form {
  display: flex;
  padding-top: 28px;
}

.app-title {
  font-size: 3rem;
  margin: 0;
  padding: 0;
}

input {
  -webkit-appearance: none;
  font-size: 1rem;
  height: 28px;
  width: 50%;
  margin: 10px 0;
}

.show-btn {
  background: #102731;
  border-style: none;
  color: #fff;
  height: 28px;
  width: 50%;
}

.show-btn:hover {
  background: #8FA7AD;
}
