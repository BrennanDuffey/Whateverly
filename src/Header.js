import React, { Component } from 'react';
import './Header.css';

class Header extends Component {
  constructor() {
    super();
    this.state = {
      showFavorites: true
    }
  }

  handleChange = (e) => {
    e.preventDefault();
    this.props.generateContent(e.target.value.trim().toLowerCase());
  }

  handleClick = (e) => {
    e.preventDefault();
    this.props.setFavoriteState(this.state.showFavorites)
    this.setState({showFavorites: !this.state.showFavorites})
  }

  render() {
    let showFavBtn = 'Show Favorites';
    if (!this.state.showFavorites) {
      showFavBtn = "Show All";
    } 

    return(
      <header>
        <h1>Kayla's Ark</h1>
        <input type="search" placeholder=" Search Animal or Country..." onChange={this.handleChange} />  
        <button onClick={this.handleClick}>{showFavBtn}</button>
      </header>
    
    )
  }
}

export default Header;
