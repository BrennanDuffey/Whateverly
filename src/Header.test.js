import React from 'react';
import Header from './Header.js';
import { shallow } from 'enzyme';


const mockGenerateContent = jest.fn();
const mocksSetFavoriteState = jest.fn();

describe('Header', () => {
  let wrapper 

  beforeEach(() => {
    wrapper = shallow(
      <Header 
        generateContent={mockGenerateContent} 
        setFavoriteState={mocksSetFavoriteState}
      />
    ); 
  });

  it('should match the snapshot with all of the data passed in', () => {
      expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
      expect(wrapper.state()).toEqual( {showFavorites: true});
  });

  it('Should call invoke handleClick when button is clicked', () => {
    expect(wrapper.state()).toEqual( {showFavorites: true});
    wrapper.find('.show-favorites').simulate('click', {
      preventDefault: () => {
      }
    });
    expect(wrapper.state()).toEqual( {showFavorites: false});
    expect(mocksSetFavoriteState).toBeCalled();
    });

  it('Should have a function handle change that fires off generateContent fn', () => {
    wrapper.find('.search').simulate('change', {
      preventDefault: () => {},
      target: { value: 'text' }
    });
    expect(mockGenerateContent).toBeCalled();
  });
});