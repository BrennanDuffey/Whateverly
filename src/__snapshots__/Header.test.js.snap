// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with all of the data passed in 1`] = `
<header>
  <h1
    className="app-title"
  >
    Kayla's Ark
  </h1>
  <input
    className="search"
    onChange={[Function]}
    placeholder=" Search Animal or Country..."
    type="search"
  />
  <button
    className="show-favorites show-btn"
    onClick={[Function]}
  >
    Show Favorites
  </button>
</header>
`;
